src/services/mock/graphql.js

```diff
@@ -0,0 +1,94 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const data = require('./json/index')
+
+/**
+ * Mapping between a query and the returned data. The returned data
+ * can be generated programmatically, or with GraphiQL, or another GraphQL
+ * command line.
+ *
+ * @type {{GscanSubscriptionQuery: {workflows: {}}, play: {data: {play: {result: [{response: (boolean|string)[], id: string}], __typename: string}}}, resume: {data: {resume: {result: [{response: (boolean|string)[], id: string}], __typename: string}}}, stop: {data: {stop: {result: [{response: (boolean|string)[], id: string}], __typename: string}}}, IntrospectionQuery: {errors: {}, data: {__schema: {queryType: {kind: string, name: string, description: string, fields: {}, inputFields: *, interfaces: {}, enumValues: *, possibleTypes: *, __typename: string}, mutationType: {kind: string, name: string, description: *, fields: {}, inputFields: *, interfaces: {}, enumValues: *, possibleTypes: *, __typename: string}, subscriptionType: {name: string, __typename: string}, types: {}, directives: {}}}}, OnWorkflowTreeDeltasData: {deltas: {added: {workflow: {id: string, name: string, status: string, owner: string, host: string, port: number, stateTotals: {waiting: number, expired: number, preparing: number, 'submit-failed': number, submitted: number, running: number, failed: number, succeeded: number}, latestStateTasks: {running: {}}, cyclePoints: {}, taskProxies: {}, familyProxies: {}}}}}, WorkflowsTableQuery: {workflows: {}}, pause: {data: {pause: {result: [{response: (boolean|string)[], id: string}], __typename: string}}}, DashboardSubscriptionQuery: {workflows: {}}}}
+ */
+const DATA_MAPPING = {
+  IntrospectionQuery: data.IntrospectionQuery,
+  OnWorkflowTreeDeltasData: {
+    deltas: {
+      added: {
+        workflow: data.OnWorkflowTreeDeltasData.workflows[0]
+      }
+    }
+  },
+  DashboardSubscriptionQuery: data.DashboardSubscriptionQuery,
+  GscanSubscriptionQuery: data.GscanSubscriptionQuery,
+  WorkflowsTableQuery: data.WorkflowsTableQuery
+}
+
+/**
+ * Given a GraphQL query, this function tries to fetch the operation
+ * name from a subscription, query, or mutation GraphQL query.
+ *
+ * Sometimes requests contain the .OperationName properties, but other
+ * times they don't, so we have to try to guess it either from the
+ * query name, or from its text (this function does the former, for
+ * introspection we are searching for __schema elsewhere).
+ *
+ * Returns an empty string if not found.
+ *
+ * @param {string} query - GraphQL query
+ * @returns {string} operation name, or empty string if not found
+ */
+function getOperationName (query) {
+  const regexp = /\s*(subscription|query|mutation)\s+([a-zA-Z0-9_]+)/
+  const match = regexp.exec(query)
+  if (match && match.length >= 3) {
+    return match[2]
+  }
+  return ''
```

Comment:
Sometimes we have the `req.body.operationName`. I think ApolloClient adds it for Query requests. But GraphiQL doesn't, and WebSockets also don't include the operation name, so we have to search for it in the text query.